middleware/proxy/provider/etcd/etcd.go

```diff
@@ -0,0 +1,165 @@
+// Package etcd is an Etcd backed provider.
+package etcd
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/coreos/etcd/client"
+	"github.com/mholt/caddy/middleware/proxy/provider"
+	"golang.org/x/net/context"
+)
+
+const (
+	// Scheme is Ectd url scheme.
+	Scheme = "etcd://"
+
+	// DefaultDirectory is the default Etcd config directory.
+	DefaultDirectory = "/caddy_proxy_hosts/"
```

Comment:
What I've seen in some software is that ownership of a default path in etcd is obtained by having ownership of a domain. In this case the software's domain is `caddyserver.com`, so it might make sense to change the default path to `/caddyserver.com/proxy/default`.